sqlx-core/src/pool/connection.rs

```diff
@@ -56,8 +56,9 @@ impl<DB: Database> DerefMut for PoolConnection<DB> {
 }
 
 impl<DB: Database> PoolConnection<DB> {
-    // explicitly release a connection from the pool
+    /// Explicitly release a connection from the pool
     pub fn release(mut self) -> DB::Connection {
+        DecrementSizeGuard::new(&self.pool);
```

Comment:
Instead of explicitly constructing a guard, you should do 
```rust
self.live.take().expect("PoolConnection double-dropped").float(&self.pool).detach()
```

and then implement `fn detach()` in this `impl`: https://github.com/launchbadge/sqlx/blob/fa7981f68a95d4cddbb07843af62b41c730ee0a3/sqlx-core/src/pool/connection.rs#L131-L163

and just have it return `self.inner.raw`